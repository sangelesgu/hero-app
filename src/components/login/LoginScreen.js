import React, { useContext } from 'react'
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const LoginScreen = ({history}) => {

    const {dispatch} = useContext(AuthContext);

    
    const addUser = {
        name: 'Samuel'
    }
    const handleLogin = () => {
        const lastPath = localStorage.getItem('lastPath') || '/'
        dispatch({
            type: types.login,
            payload: addUser
        }); 

        history.replace(lastPath); 
        
    }
    return (
        <div className="container mt-5">
            <h1>Login</h1>
            <hr/>

            <button
             className="btn btn-primary"
             onClick={handleLogin}
            >
            Login 
            </button>
        </div>
    )
}
