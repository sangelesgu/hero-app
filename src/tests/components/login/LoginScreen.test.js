import React from 'react';
import { mount } from "enzyme";
import { LoginScreen } from "../../../components/login/LoginScreen";
import { AuthContext } from '../../../auth/AuthContext';

describe('Test in <LoginScreen />', () => {
  const history = {
    replace: jest.fn()
  }
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false
    }
  }

  const wrapper = mount (
    <AuthContext.Provider value={contextValue}> 
      <LoginScreen history={history} />
    </AuthContext.Provider>
  );
  
  test('should return snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  }); 

  test('should do dispatch and navigate', () => {
    
  });
  
  
})
