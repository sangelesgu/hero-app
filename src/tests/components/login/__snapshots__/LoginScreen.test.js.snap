// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LoginScreen /> should return snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
